src/app/services/actions/snippet-action-creator.ts

```diff
@@ -56,13 +58,17 @@ export function getSnippet(language: string): Function {
 
       const requestBody =
         sampleQuery.sampleBody &&
-          Object.keys(sampleQuery.sampleBody).length !== 0 && // check if empty object
-          sampleQuery.sampleBody.trim() !== ''
+          Object.keys(sampleQuery.sampleBody).length !== 0
           ? JSON.stringify(sampleQuery.sampleBody)
           : '';
 
-      // eslint-disable-next-line max-len
-      const body = `${sampleQuery.selectedVerb} /${queryVersion}/${requestUrl + search} HTTP/1.1\r\nHost: graph.microsoft.com\r\nContent-Type: application/json\r\n\r\n${requestBody}`;
+      if(sampleQuery.selectedVerb === 'GET'){
+        // eslint-disable-next-line max-len
+        body = `${sampleQuery.selectedVerb} /${queryVersion}/${requestUrl + search} HTTP/1.1\r\nHost: graph.microsoft.com\r\nContent-Type: application/json\r\n\r\n`;
+      } else {
+        // eslint-disable-next-line max-len
+        body = `${sampleQuery.selectedVerb} /${queryVersion}/${requestUrl + search} HTTP/1.1\r\nHost: graph.microsoft.com\r\nContent-Type: application/json\r\n\r\n${requestBody}`;
+      }
```

Comment:
The body string described when the verb is GET looks similar to what is in the next section except for the addition of the `requestBody` variable.

```suggestion
        // eslint-disable-next-line max-len
        const body: string = `${sampleQuery.selectedVerb} /${queryVersion}/${requestUrl + search} HTTP/1.1\r\nHost: graph.microsoft.com\r\nContent-Type: application/json\r\n\r\n`;
       
      if (sampleQuery.selectedVerb !== 'GET') {
        body += `${requestBody}`;
      }
```

Reply:
Yeah, a GET request shouldn't have a request body...
I am working on refactoring this though and seeing how we can pass more sampleHeaders... and not hardcoding the Content-Type...